### ISSUE TITLE
Tr1d1um hook api is always overriding Duration provided by user

### ISSUE THREAD

**Issue by Sachin4403** [xmidt-org/webpa-common#565]
# Version
```  
tr1d1um: v0.5.2
```

# The problem 
Code written in [xwebhook/transport.go](https://github.com/xmidt-org/webpa-common/blob/main/xwebhook/transport.go#L155-L156) is always overriding the duration field  with the default value

```golang
        // always set duration to default
	webhook.Duration = defaultWebhookExpiration
```
Ideally, it should be if the user is not providing the duration then it should override.

**Comment by joe94** [comment]
Hey @Sachin4403, we're hoping to improve this but not until our next API breaking change. Notice this default is only applied when the `Until` field is not provided https://github.com/xmidt-org/webpa-common/blob/7740b009eb2cada45954289240d73626e82ccb0d/xwebhook/transport.go#L158

So you can pass in that field to specify when your webhook will expire.
You can use the RFC3339 format for the time in your JSON: ```2006-01-02T15:04:05Z```

**Comment by Sachin4403** [comment]
Hi @joe94 
Here we are overriding Duration
https://github.com/xmidt-org/webpa-common/blob/7740b009eb2cada45954289240d73626e82ccb0d/xwebhook/transport.go#L156
 and using the same duration to set the TTL for DB
https://github.com/xmidt-org/webpa-common/blob/b98c7113262ce7f2879f60dee1fd0df2923dc003/xwebhook/service.go#L82-L87 
[here](https://github.com/xmidt-org/argus/blob/d8392c08124bf1d535e51790305d4a87c8bb6e09/store/cassandra/executer.go#L65) record will be saved in DB
Now in DB current record will be deleted after TTL is passed(i.e 5 minutes and until will be valid for the next 24 hours.).

# The problem starts now
Now if I start a new caduceus(due to the high load on the old instance), I will not have the existing webhook and  I will be losing the events which are coming to the new caduceus instance. we will have still very high load on existing caduceus instance.

Note: New Caduceus instance will be autoscaled automatically. if it was in our hand then we can reregister the webhook unfortunately it is not in our hand.

**Comment by joe94** [comment]
By "old" and "new" caduceus, do you mean caduceus that stores webhooks through SNS and then Argus respectively?

Presumably if your caduceus machines are all on Argus already, the DB should be shared and the new nodes have access to that DB.

**Comment by Sachin4403** [comment]
By `OLD` I mean an EC2 instance that is running already and had picked the webhook from DB and currently sending the evets to webhooks(senders).

By `NEW` i mean if the OLD EC2 instance was not able to handle the load fully then autoscaling will happen in the cloud, then the New caduceus instance will be deployed automatically. New Caduceus will try to poll the webhooks from argus 
1. but it won't get any webhook.
2. Any calls coming to this caduceus instance will drop the WRPMessage as there are no senders(webhooks) for this caduceus instance.

Note: We are using Argus with yugabyte DB

**Comment by joe94** [comment]
Hey @Sachin4403, thanks for sharing that. I created a separate issue where we can focus on talking about this caduceus issue: https://github.com/xmidt-org/caduceus/issues/258

As for the original issue regarding the duration, is there anything else you had in mind before I close this? FYI We moved `xmidt-org/webpa-common/xwebhook` to `xmidt-org/ancla` and we already started having discussions on how to go about improve the webhook API there https://github.com/xmidt-org/ancla/issues/15

**Comment by Sachin4403** [comment]
Hi, @joe94  there is no issue with the caduceus, Caduceus will work with webhook which Argus is responsible for providing it.
I had created a sequence diagram of it to explain it better. this `Duration` is the one who is creating the problem.
Due to this Duration  caduceus, Instance No 2 will not get any webhooks 
![issue](https://user-images.githubusercontent.com/12969957/110322114-a33add80-8038-11eb-9172-ba6e68ce316e.png)

And i saw in <b>ancla</b>  you are overriding  the Duration.
https://github.com/xmidt-org/ancla/blob/103b6b2de50e1116c08e2ae1c8fa537918086310/transport.go#L174-L175

**Comment by joe94** [comment]
Ah, gotcha. That makes sense @Sachin4403. Thanks for the detailed explanation! That's definitely something we can fix in ancla and I created the issue to track it https://github.com/xmidt-org/ancla/issues/44

**Comment by joe94** [comment]
For now what you can do is re-register your webhooks within the 5 mins to keep your registration list active.

**Comment by Sachin4403** [comment]
Thanks, @joe94  for fixing this issue in ancla,  Going forward will use the latest version of Tr1d1um which will have ancla instead of the xwebhook module.
